src/nomad_simulations/numerical_settings.py

```diff
@@ -413,33 +364,239 @@ def normalize(self, archive, logger) -> None:
+    A base section used to define the settings of a k-line path within a multidimensional mesh.
     """
 
+    high_symmetry_path = Quantity(
+        type=JSON,
+        description="""
+        Dictionary containing the high-symmetry points (in units of the `reciprocal_lattice_vectors`) followed in
+        the k-line path. E.g., in a cubic lattice:
+            high_symmetry_path = {
+                'Gamma': [0, 0, 0],
+                'X': [0.5, 0, 0],
+                'Y': [0, 0.5, 0],
+            }
+        """,
+    )
+
+    n_line_points = Quantity(
+        type=np.int32,
+        description="""
+        Number of points in the k-line path.
+        """,
+    )
+
     points = Quantity(
-        type=np.complex128,
-        shape=['n_points', 'dimensionality'],
-        unit='joule',
+        type=np.float64,
+        shape=['n_line_points', 3],
         description="""
-        List of all the points in the mesh in joules.
+        List of all the points in the k-line path in units of the `reciprocal_lattice_vectors`.
         """,
     )
 
+    def get_high_symmetry_points_norm(
+        self,
+        reciprocal_lattice_vectors: Optional[pint.Quantity],
+    ) -> Optional[dict]:
+        """
+        Get the high symmetry points norms from the dictionary of vectors in units of the `reciprocal_lattice_vectors`. This
+        function is useful when matching lists of points passed as norms to the high symmetry points in order to resolve
+        `KLinePath.points`
+
+        Args:
+            reciprocal_lattice_vectors (Optional[np.ndarray]): The reciprocal lattice vectors of the atomic cell.
+
+        Returns:
+            (Optional[dict]): The high symmetry points norms.
+        """
+        # Checking if `reciprocal_lattice_vectors` is defined and taking its magnitude to operate
+        if reciprocal_lattice_vectors is None:
+            return None
+        rlv = reciprocal_lattice_vectors.magnitude
+
+        # initializing the norms dictionary
+        high_symmetry_points_norms = {
+            key: 0.0 * reciprocal_lattice_vectors.u
+            for key in self.high_symmetry_path.keys()
+        }
+        # initializing the first point
+        prev_value_norm = 0.0 * reciprocal_lattice_vectors.u
+        prev_value_rlv = np.array([0, 0, 0])
+        for i, (key, value) in enumerate(self.high_symmetry_path.items()):
+            if i == 0:
+                continue
+            value_rlv = value @ rlv
+            value_tot_rlv = value_rlv - prev_value_rlv
+            value_norm = (
+                np.linalg.norm(value_tot_rlv) * reciprocal_lattice_vectors.u
+                + prev_value_norm
+            )
+            high_symmetry_points_norms[key] = value_norm
+
+            # accumulate value vector and norm
+            prev_value_rlv = value_rlv
+            prev_value_norm = value_norm
+        return high_symmetry_points_norms
```

Comment:
These kinds of maniuplations are more succint using the functional programming paradigm:

```python
import numpy as np
import operator
from itertools import tee

def get_hsp_norm(self):
  hsp = (lambda x: x + [x[0]])(list(self.high_symmetry_path.values()))  # make hsp path circular
  hsp_prev, hsp_next = tee(map(lambda x: rlv @ x, hsp))  # generate 2 iterators of the corrected hsp
  next(hsp_next, None)  # shift 1 iterator ahead
  deltas = map(operator.sub, hsp_next, hsp_prev)  # compute the differences
  return list(accumulate(map(np.linalg.norm, deltas), func=operator.add))  # acrue the norms
```